kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/appliers/reports/ReportsVariantApplier.kt

```diff
@@ -135,6 +144,14 @@ internal sealed class ReportsVariantApplier(
             runOnCheck += xmlTask
         }
 
+        icTask.configure {
+            icFile.convention(reportConfig.ic.icFile)
+            filters.set((reportConfig.ic.filters ?: reportConfig.filters ?: commonFilters).convert())
+        }
+        runOnCheck += reportConfig.ic.onCheck.map { run ->
```

Comment:
Why we have to do `map` here while `reportConfig.xml.onCheck` can be accessed directly?

Reply:
Because a lazy configuration is used here, which is based on the use of providers.

Gradle [recommends](https://docs.gradle.org/current/userguide/lazy_configuration.html#creating_property_provider) using the `Provider#map` function to create them.

> You should prefer using map() instead, as this has some useful benefits, which we will see later.

In fact, this has many advantages.

Reply:
Why `reportConfig.xml.onCheck` is not lazy then?